### PR TITLE
feat: use sass instead of node-sass

### PR DESCRIPTION
**Pull request by Nek-**
| Q               | A
| --------------- | -----
| Branch?         | master (could be done on others, your call)
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | yes (it won't work with gulp-sass 4.0 anymore I think)
| Deprecations?   | no/yes 
| Related tickets | #12893 
| License         | MIT

node-sass is deprecated and its installation breaks on many many setups depending on:
- Node version (up to 12 it seems fine, it breaks for 13+)
- gpy installed or not
- Already installed tooling
- Platform (linux/mac/windows)

All of these problems make the `sass` package a benediction. This commit makes the sylius codebase compatible with it.

### CODE REVIEW

**Review by lchrusciel — COMMENTED** [review]
I assume, that it will require update of package.json on the end developers apps. Can you add a note to UPGRADE.md?

**Comment by Nek-** [comment]
@lchrusciel done!

**Comment by lchrusciel** [comment]
Hey Maxime, to no hold you without answer, I've requested a review from @kulczy, as he is our main frontend guy. Next week I hope we will be ready to merge it :)

**Comment by joshuabyler** [comment]
having this would be perfect to help out with updating node versions

**Comment by lchrusciel** [comment]
it is still quite high on our list, we didn't forget about you

**Comment by Nek-** [comment]
Rebased!

**Review by kulczy — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Maxime! :1st_place_medal:

**Comment by Nek-** [comment]
@lchrusciel why isn't this merged in 1.11 ? 🤔 Do you need another PR ?

**Comment by lchrusciel** [comment]
Hey Maxime,

I'm reluctant to js deps in the patch releases. If it would land in 1.11, after the update all users would have to change theirs package.json. Patch releases are considered as safer than minor releases from my experience, so I would prefer to leave it to a more conscious update - which is how I preserve minor updates.

**Comment by porl** [comment]
Hi Ichrusciel

Does this mean it won't be in any release until 1.12? Or will there be a 1.11.x release with it?

I am currently on 1.10 and looking at building a newer site. I will hold off until 1.12 if this is the case so I don't need to manually patch things for my newer server (running Node 17).

**Comment by Nek-** [comment]
@porl it will come with 1.12, not before, sorry.

**Comment by lchrusciel** [comment]
Hey folks, Sylius v1.12 is not yet scheduled, but we will see if it can be sooner than later. 

We are looking for someone who will help us provide an anti-corruption layer to preserve backward compatibility with this task, but that may be hard as we are serving js together with the main Sylius codebase. If you have any advice, I'm open hearing them :)

**Pull request merged**
